### PR TITLE
Feature/user friend list

### PR DESCRIPTION
**Pull request by ArnaudPoletto**
Users now have a list of friend ids as an attribute. Can also be seen in the database.

### CODE REVIEW

File: app/src/main/java/sdp/moneyrun/user/User.java [hunk 1]
```diff
@@ -30,228 +38,280 @@ public void setMaxScoreInGame(int maxScoreInGame) {
...
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        User user = (User) o;
+        return userId.equals(user.userId) &&
+                numberOfPlayedGames == user.numberOfPlayedGames &&
+                numberOfDiedGames == user.numberOfDiedGames &&
+                name.equals(user.name) &&
+                address.equals(user.address);
+    }
```
Comment (RafaelRamirezS) [inline app/src/main/java/sdp/moneyrun/user/User.java:287]: Do you have to add the friends list to the .equals() criteria ? I'm not sure, but I wanted to check with you

**Review by RafaelRamirezS — CHANGES_REQUESTED** [review]

File: app/src/main/java/sdp/moneyrun/user/User.java [hunk 2]
```diff
@@ -1,25 +1,33 @@
...
-        private int numberOfDiedGames;
-        private int totalDistanceRun;
+    private String userId;
+    private String name;
+    private String address;
+    private int numberOfPlayedGames;
+    private int numberOfDiedGames;
+    private int totalDistanceRun;
+    // By default, user has no friend. But we still add a default invalid id
+    // to make the node appear in the database.
```
[the remark this answers is not in the document]
Comment (ArnaudPoletto) [reply app/src/main/java/sdp/moneyrun/user/User.java:23]: I can understand that. I first went with that implementation but decided to remove the default value, I just forgot to remove the comment.

**Review by Warhedgehog — APPROVED** [review]
Except from what is pointed out in the other review, this looks good to me

File: app/src/main/java/sdp/moneyrun/user/User.java (hunk 1 above)
Comment (ArnaudPoletto) [reply app/src/main/java/sdp/moneyrun/user/User.java:286]: Yes, this is important, thank you

Comment (ArnaudPoletto) [reply app/src/main/java/sdp/moneyrun/user/User.java:286]: done

**Review by RafaelRamirezS — APPROVED** [review]

**Pull request merged**